src/apps/basic/basic_apps.lua

```diff
@@ -95,20 +95,14 @@ function Tee:new ()
 end
 
 function Tee:push ()
-   noutputs = #self.output
-   if noutputs > 0 then
-      local maxoutput = link.max
-      for _, o in ipairs(self.output) do
-         maxoutput = math.min(maxoutput, link.nwritable(o))
-      end
-      for _, i in ipairs(self.input) do
-         for _ = 1, math.min(link.nreadable(i), maxoutput) do
-            local p = receive(i)
-            maxoutput = maxoutput - 1
-            do local output = self.output
-               for k = 1, #output do
-                  transmit(output[k], k == #output and p or packet.clone(p))
-               end
+   for _, inport in ipairs(self.input) do
+      while not inport:empty() do
+         local pkt = inport:receive()
+         local used = false
+         for _, outport in ipairs(self.output) do
+            if not outport:full() then
+               outport:transmit(used and pkt:clone() or pkt)
+               used = true
```

Comment:
Is this a rewrite of the `Tee` app to preserve the functionality but make the code shorter? That would be very welcome: nothing is better than reducing code size while preserving functionality :-).

Review is a bit hard though when there is no commit message explaining why the change is being made. Since this is included with the commit that adds the Virtio-net driver I am left wondering whether this is actually modifying the `Tee` behavior in some way that the Virito-net driver depends on.

(Just as feedback for the future: I'm not suggesting to go back and rebase it now.)